butane_core/src/migrations/fsmigrations.rs

```diff
@@ -135,16 +145,74 @@ impl FsMigration {
     fn lock_shared(&self) -> Result<MigrationLock> {
         MigrationLock::new_shared(&self.root.join("lock"))
     }
+
+    /// Delete all of the files except info.json which is recreated
+    /// with only `from_name` set to allow migration series traversal.
+    pub fn delete_db(&self) -> Result<()> {
+        let entries = self.fs.list_dir(&self.root)?;
+        for entry in entries {
+            match entry.file_name() {
+                None => continue,
+                Some(name) => {
+                    let name = name.to_string_lossy();
+                    if name == "info.json" {
+                        // Re-create info.json using the minimum required to allow
+                        // `all_migrations` to traverse the list.
+                        let info = self.info()?;
+                        let info = MigrationInfo {
+                            from_name: info.from_name,
+                            ..Default::default()
+                        };
+                        self.write_info(&info)?;
+                    } else {
+                        self.fs.delete(&entry)?;
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
 }
 
 impl MigrationMut for FsMigration {
-    fn write_table(&mut self, table: &ATable) -> Result<()> {
+    fn add_modified_table(&mut self, table: &ATable) -> Result<()> {
         self.write_contents(
             &format!("{}.table", table.name),
             serde_json::to_string_pretty(table)?.as_bytes(),
         )
     }
 
+    fn add_unmodified_table(
+        &mut self,
+        table: &ATable,
+        from_migration: &impl Migration,
+    ) -> Result<()> {
+        // It isnt possible to use from_migration.info() as that isn't part of `Migration` trait.
+        // So first step is to convert to FsMutation.
+        let migrations_dir = self.root.parent().unwrap();
+        let migrations = crate::migrations::from_root(migrations_dir);
+        let migration_list = migrations.all_migrations()?;
+        let from_mutation = migration_list
```

Comment:
[nit] I don't love using the name "mutation" for a variable which represents a migration. I think you're doing it because there's already an existing variable named `from_migration`. I'd prefer doing one of the following
1. Assign the name to another intermediate variable and then alias `from_migration`
2. Consider a name such as `from_fsmigration` which imo better conveys the difference between the two variables (one is an `FsMigration` instead of just a `Migration`